core/chains/evm/logpoller/log_poller.go

```diff
@@ -319,55 +320,50 @@ func convertTopics(topics []common.Hash) [][]byte {
+// block range [start, end] and save them to the db.
+// Retries until ctx cancelled. Will return an error if cancelled
+// or if there is an error backfilling.
+func (lp *logPoller) backfill(ctx context.Context, start, end int64) error {
 	for from := start; from <= end; from += lp.backfillBatchSize {
-		var (
-			logs []types.Log
-			err  error
-		)
 		to := mathutil.Min(from+lp.backfillBatchSize-1, end)
-		// Retry forever to query for logs,
-		// unblocked by resolving node connectivity issues.
-		utils.RetryWithBackoff(ctx, func() bool {
-			logs, err = lp.ec.FilterLogs(ctx, lp.filter(big.NewInt(from), big.NewInt(to), nil))
-			if err != nil {
-				lp.lggr.Warnw("Unable query for logs, retrying", "err", err, "from", from, "to", to)
-				return true
-			}
-			return false
-		})
+		logs, err := lp.ec.FilterLogs(ctx, lp.filter(big.NewInt(from), big.NewInt(to), nil))
+		if err != nil {
+			lp.lggr.Warnw("Unable query for logs, retrying", "err", err, "from", from, "to", to)
+			return err
+		}
 		if len(logs) == 0 {
 			continue
 		}
 		lp.lggr.Infow("Backfill found logs", "from", from, "to", to, "logs", len(logs))
-		// Retry forever to save logs,
-		// unblocked by resolving db connectivity issues.
-		utils.RetryWithBackoff(ctx, func() bool {
-			err = lp.orm.q.WithOpts(pg.WithParentCtx(ctx)).Transaction(func(tx pg.Queryer) error {
-				return lp.orm.InsertLogs(convertLogs(lp.ec.ChainID(), logs), pg.WithQueryer(tx))
-			})
-			if err != nil {
-				lp.lggr.Warnw("Unable to insert logs logs, retrying", "err", err, "from", from, "to", to)
-				return true
-			}
-			return false
+		err = lp.orm.q.WithOpts(pg.WithParentCtx(ctx)).Transaction(func(tx pg.Queryer) error {
+			return lp.orm.InsertLogs(convertLogs(lp.ec.ChainID(), logs), pg.WithQueryer(tx))
 		})
+		if err != nil {
+			lp.lggr.Warnw("Unable to insert logs logs, retrying", "err", err, "from", from, "to", to)
+			return err
+		}
 	}
-	return end + 1
+	return nil
 }
 
 // getCurrentBlockMaybeHandleReorg accepts a block number
 // and will return that block if its parent points to our last saved block.
+// One can optionally pass the block header if it has already been queried to avoid an extra RPC call.
 // If its parent does not point to our last saved block we know a reorg has occurred,
 // so we:
 // 1. Find the LCA by following parent hashes.
 // 2. Delete all logs and blocks after the LCA
 // 3. Return the LCA+1, i.e. our new current (unprocessed) block.
-func (lp *logPoller) getCurrentBlockMaybeHandleReorg(ctx context.Context, currentBlockNumber int64) (*types.Header, error) {
-	currentBlock, err1 := lp.ec.HeaderByNumber(ctx, big.NewInt(currentBlockNumber))
-	if err1 != nil {
-		lp.lggr.Warnw("Unable to get currentBlock", "err", err1, "currentBlockNumber", currentBlockNumber)
-		return nil, err1
+func (lp *logPoller) getCurrentBlockMaybeHandleReorg(ctx context.Context, currentBlockNumber int64, currentBlock *types.Header) (*types.Header, error) {
+	var err1 error
```

Comment:
`err1` is only used within the if block below, can't we declare it in the block?